collections/map.go

```diff
@@ -127,7 +127,7 @@ func (m Map[K, V]) Iterate(ctx context.Context, ranger Ranger[K]) (Iterator[K, V
 // walk function with the decoded key and value. If the callback function
 // returns true then the walking is stopped.
 // A nil ranger equals to walking over the entire key and value set.
-func (m Map[K, V]) Walk(ctx context.Context, ranger Ranger[K], walkFunc func(K, V) bool) error {
+func (m Map[K, V]) Walk(ctx context.Context, ranger Ranger[K], walkFunc func(key K, value V) (stop bool, err error)) error {
```

Comment:
added the possibility to make `walkFunc` error, this plays well with gov and generally is better to allow for a straightforward way to error during iterations and propagate errors back.